### PR TITLE
Spyder support

### LINKED ISSUE #15492: Add spyder IDE support to github desktop
### The feature request

Add support to add spyder python IDE as GitHub desktop's default IDE

### Proposed solution

The benefit of it will be that users who program python using spyder IDE and github

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by davidlloyduk**
Closes #15492

## Description

 - windows support
 - mac support
 - linux support

### Screenshots

spyder1
spyder2
spyder3
spyder4

## Release notes

Notes: 

- Adds support for Spyder IDE integration on Windows, Mac and Linux

### CODE REVIEW

File: app/src/lib/editors/launch.ts [hunk 1]
```diff
@@ -30,7 +30,9 @@ export async function launchExternalEditor(
   }
 
   if (editor.usesShell) {
-    spawn(`"${editorPath}"`, [`"${fullPath}"`], { ...opts, shell: true })
+    //wrap each shell argument in quotes as it may include paths with spaces 
+    const args = editor.shellArgs ? editor.shellArgs?.map(arg => `"${arg}"`) : []
```
Comment (davidlloyduk) [inline app/src/lib/editors/launch.ts:34]: have assumed here that all arguments are paths, will need to rethink this as arguments maybe other things e.g integer - might have to wrap them when they are defined.

File: app/src/lib/editors/darwin.ts [hunk 2]
```diff
@@ -144,6 +144,10 @@ const editors: IDarwinExternalEditor[] = [
     name: 'Lite XL',
     bundleIdentifiers: ['com.lite-xl'],
   },
+  {
+    name: 'Spyder',
```
Comment (davidlloyduk) [inline app/src/lib/editors/darwin.ts:148]: happy to remove this if you wish to use [PR 15505](https://github.com/desktop/desktop/pull/15505)

File: app/src/lib/editors/launch.ts [hunk 3]
```diff
@@ -30,7 +30,8 @@ export async function launchExternalEditor(
   }
 
   if (editor.usesShell) {
-    spawn(`"${editorPath}"`, [`"${fullPath}"`], { ...opts, shell: true })
+    const args = editor.shellArgs ? editor.shellArgs : []
```
Comment (sergiou87) [inline app/src/lib/editors/launch.ts]: This can be simplified as:
```suggestion
    const args = editor.shellArgs ?? []
```

**Review by sergiou87 — CHANGES_REQUESTED** [review]
Thank you for your contribution! This makes total sense to me 😄 

Only found one thing that could be simplified, once that's fixed I'll try it out and merge it if all goes well.

**Comment by sergiou87** [comment]
@davidlloyduk seems like you need to fix linting. I think you can use `yarn lint:fix` from the terminal and it will fix the source files affected.

**Comment by davidlloyduk** [comment]
Will sort out the linting! Thanks

**Comment by ChadMichaelSheda** [comment]
Thanks go for it

File: app/src/lib/editors/launch.ts (hunk 3 above)
Comment (davidlloyduk) [reply app/src/lib/editors/launch.ts]: sorted in latest commit

**Review by sergiou87 — CHANGES_REQUESTED** [review]
On Windows at least, this seems to work for individual files, but not to open the whole repository, where I get this error:

image

Also, I wasn't able to run it on macOS at all. I cannot open Spyder independently either 😞  Could you maybe record a video of it working? (Opening both individual files and a whole repository)

Thanks!

**Comment by davidlloyduk** [comment]
> On Windows at least, this seems to work for individual files, but not to open the whole repository, where I get this error:
> 
> image
> 
> Also, I wasn't able to run it on macOS at all. I cannot open Spyder independently either 😞 Could you maybe record a video of it working? (Opening both individual files and a whole repository)
> 
> Thanks!

will investigate and sort out the directory thing, bah to the mac thing - copied it from that other guys PR - will fire up a vm and test the mac

**Comment by niik** [comment]
Hey @davidlloyduk, since we haven't heard from you in a while and since this PR has gone stale with conflicts I'm going to close this out to keep our PR list somewhat manageable. If you get back to this feel free to open a new PR!

**Comment by TessaNM** [comment]
200568001-11283616-d9bf-498f-bdf1-843c905e6b8c
In Github desktop you have to include spyder -p in the arguments box (before %TARGET_PATH%) after specifying the path to Spyder IDE - I was able to open the repository in Spyder IDE when I did that.

**Pull request closed without merging**